apis/deployer/healthchecks/validation/validation_test.go

```diff
@@ -0,0 +1,254 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/gardener/landscaper/apis/deployer/healthchecks"
+	"github.com/gardener/landscaper/apis/deployer/healthchecks/validation"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+)
+
+func TestConfig(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "validation Test Suite")
+}
+
+var _ = Describe("Validation", func() {
+
+	var (
+		ctx context.Context
+		fld *field.Path
+
+		hc healthchecks.HealthChecksConfiguration
+		cc healthchecks.CustomHealthCheckConfiguration
+		rc healthchecks.RequirementSpec
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		fld = field.NewPath("healthcheck")
+
+		rc = healthchecks.RequirementSpec{
+			JsonPath: "{ .foo.bar }",
+			Operator: selection.Equals,
+			Value: []runtime.RawExtension{
+				{
+					Raw: []byte("foobar"),
+				},
+			},
+		}
+
+		cc = healthchecks.CustomHealthCheckConfiguration{
+			Name:     "customHealthCheck",
+			Timeout:  nil,
+			Disabled: false,
+			Resource: &lsv1alpha1.TypedObjectReference{
+				APIVersion: "v1",
+				Kind:       "Service",
+				ObjectReference: lsv1alpha1.ObjectReference{
+					Name:      "foo",
+					Namespace: "bar",
+				},
+			},
+			LabelSelector: &healthchecks.LabelSelectorSpec{
+				APIVersion: "v1",
+				Kind:       "Service",
+				Labels: map[string]string{
+					"foo": "bar",
+				},
+			},
+			Requirements: []healthchecks.RequirementSpec{rc},
+		}
+
+		hc = healthchecks.HealthChecksConfiguration{
+			DisableDefault:     true,
+			Timeout:            &lsv1alpha1.Duration{Duration: 180 * time.Second},
+			CustomHealthChecks: []healthchecks.CustomHealthCheckConfiguration{cc},
+		}
+
+	})
+
+	AfterEach(func() {
+		defer ctx.Done()
+	})
```

Comment:
not needed. (should be also removed in the other tests)

Reply:
Fixed.